jme3-terrain/src/main/java/com/jme3/terrain/geomipmap/TerrainPatch.java

```diff
@@ -799,8 +799,7 @@ protected void setLodBottom(int lodBottom) {
     @Override
     public int collideWith(Collidable other, CollisionResults results) throws UnsupportedCollisionException {
         if (refreshFlags != 0)
-            throw new IllegalStateException("Scene graph must be updated" +
-                                            " before checking collision");
+            System.err.println("Scene graph must be updated before checking collision");

```

Comment:
Can you use Logger instead of `System.err.println`?
Also, I think we should immediately return 0 when if the refreshFlags are modified.

Reply:
I added logger code following the same format of other JME classes, let me know if it looks alright or if I should change anything.

I also added brackets around the if statements to make the code cleaner and easier to understand, hopefully this change is okay as well.

Reply:
>I also added brackets around the if statements

Please fix the formatting issues. There should be a whitespace added in '){' and `if(`

Reply:
also in `}else`

Reply:
It should be formatted correctly now

Reply:
thanks